app/api/userData/route.js

```diff
@@ -19,15 +19,40 @@ const db = getFirestore(app);
 export async function GET(req) {
     const currentURL = new URL(req.url);
     const action = currentURL.searchParams.get('action');
-    const collection = currentURL.searchParams.get('collection');
+    const col = currentURL.searchParams.get('collection');
     const document = currentURL.searchParams.get('document');
     try {
         if (action === 'getuser') {
-            const docRef = doc(db, collection, document);
+            const docRef = doc(db, col, document);
             const docSnap = await getDoc(docRef);
             const userData = docSnap.data();
             return NextResponse.json(userData, { status: 200 })
         }
+        if (action === 'getpurchases') {
+            const allData = await getDocs(collection(db, 'store'));
```

Comment:
I agree that this function is difficult to read because of the complex queries. Do you suggest we switch to a relational db? Idk how much time it might take but it would be a good practice because I have not worked with this type of dbs yet.